docs/content/sips/xxx-trigger-executors.md

```diff
@@ -0,0 +1,60 @@
+title = "SIP xxx - Trigger Executors"
+template = "main"
+
+---
+
+Summary: A new architecture for Spin trigger executors.
+
+Owner: lann.martin@fermyon.com
+
+Created: April 29, 2022
+
+<!-- Updated: April 4, 2022 -->
+
+## Background
+
+Spin currently includes built-in implementations for two trigger types: HTTP, and Redis (PubSub).
+We want to allow for additional trigger types in the future, whether they are "internal" types developed and
+supported within the Spin codebase, or "external" types developed by the broader community.
+
+While it is currently possible to build external trigger types - as in the
+[TimerTrigger example](../../../examples/spin-timer/src/main.rs) - it requires replicating much of the
+functionality provided by the `spin up` command.
+
+## Proposal
+
+### Trigger types are implemented as "trigger executor" CLI commands:
+
+* "Internal" triggers use `spin` subcommands, e.g. `spin trigger http`
+* "External" triggers use external binaries, e.g. `spin-trigger-timer`
+* Trigger executors share common code, including common CLI options in a `spin-trigger` crate
+
+### The `spin up` command's responsibilities are reduced
+
+* Detect the trigger type for an application
+* Find a trigger executor for that type (e.g. check the `PATH` for `"spin-trigger-${type}"`)
+* `exec` the trigger executor, forwarding its own CLI arguments
+
+## Future design options
+
+_This section contains possible future features which are not fully defined here._
+
+### Multi-trigger orchestration
+
+If multiple trigger types ere to be supported by a single application, the role of `spin up` could
```

Comment:
typo: _ere_

Reply:
Wow that was fast. 😂

Reply:
Says the guy who submitted a review 2 minutes after the PR was created...